test/e2e/systeminternaltls/system_internal_tls_test.go

```diff
@@ -113,45 +107,162 @@ func TestSystemInternalTLS(t *testing.T) {
+
+	// Create ConfigMap with networking.knative.dev/trust-bundle label in required namespaces
+	cm := &corev1.ConfigMap{
+		ObjectMeta: v1.ObjectMeta{
+			Name: "knative-bundle",
+			Labels: map[string]string{
+				networking.TrustBundleLabelKey: "true",
+			},
+		},
+		Data: map[string]string{
+			"cert.pem":          string(secret.Data[certificates.CertName]),
+			"cert_reloaded.pem": string(secretReloaded.Data[certificates.CertName]),
+		},
+	}
+	for _, ns := range bundleNamespaces {
+		_, err := clients.KubeClient.CoreV1().ConfigMaps(ns).
+			Create(context.Background(), cm, v1.CreateOptions{})
+		if err != nil {
+			t.Fatal("Failed to create configmap:", err)
+		}
 	}
 
-	scanner := bufio.NewScanner(buf)
-	for scanner.Scan() {
-		if matcher(scanner.Text()) {
-			matchCount++
+	// Clean up on test failure or interrupt
+	test.EnsureCleanup(t, func() {
+		test.TearDown(clients, &names)
+		for _, ns := range bundleNamespaces {
+			if err := clients.KubeClient.CoreV1().ConfigMaps(ns).
+				Delete(context.Background(), cm.Name, v1.DeleteOptions{}); err != nil && !errors.IsNotFound(err) {
+				t.Error("Failed to delete configmap:", err)
+			}
 		}
+	})
+
+	t.Log("Deleting Secret in user namespace that is mounted by queue-proxy")
+	if err := clients.KubeClient.CoreV1().Secrets(test.ServingFlags.TestNamespace).Delete(context.Background(), pkgNetworking.ServingCertName, v1.DeleteOptions{}); err != nil {
+		t.Error(err)
+	}
+
+	t.Log("Checking queue-proxy logs")
+	pods, err := clients.KubeClient.CoreV1().Pods(test.ServingFlags.TestNamespace).List(context.TODO(), v1.ListOptions{
+		LabelSelector: fmt.Sprintf("serving.knative.dev/configuration=%s", names.Config),
+	})
+	if err != nil {
+		t.Fatal(err)
 	}
+	if len(pods.Items) == 0 {
+		t.Fatal("No pods detected for test app:", err)
+	}
+	// The Certs are loaded during startup and then after re-creating the secret.
+	const numMatches = 2
+	helloWorldPod := pods.Items[0]
+	if err := e2e.WaitForLog(t, clients, helloWorldPod.Namespace, helloWorldPod.Name, "queue-proxy", matchCertReloadLog, numMatches); err != nil {
+		t.Fatal("Certificate not reloaded in time by queue-proxy:", err)
+	}
+
+	checkEndpointState(t, clients, url)
 
-	if err = scanner.Err(); err != nil {
-		err = fmt.Errorf("failed scanning activator logs: %w", err)
-		return
+	t.Log("Deleting secret in system namespace")
+	if err := clients.KubeClient.CoreV1().Secrets(systemNS).Delete(context.Background(), config.ServingRoutingCertName, v1.DeleteOptions{}); err != nil {
+		t.Error(err)
 	}
+	_, err = waitForCaCert(clients, systemNS, config.ServingRoutingCertName)
+	if err != nil {
+		t.Fatal(err)
+	}
+	checkEndpointState(t, clients, url)
+
+	t.Log("Deleting secret in ingress namespace")
+	if err := clients.KubeClient.CoreV1().Secrets(ingressNS).Delete(context.Background(), config.ServingRoutingCertName, v1.DeleteOptions{}); err != nil {
+		t.Error(err)
+	}
+	_, err = waitForCaCert(clients, ingressNS, config.ServingRoutingCertName)
```

Comment:
In theory we should not be using the 'ca.crt' part of the secret. This only works with the CA type issuer anyway. It's just in the code to make transition to the new solution easier. Ideally we always have the trust-bundle in place and just update the CA to match whatever is used by cert-manager. So basically:

- Have a trust-bundle (could even go to the static test resources IMHO) with CA-1
- Update it in this test with the CA-2
- Rotate secrets via deleting (as is)
- Remove CA-1
- Verify that it still works

Reply:
OK. Good to know. Thanks. I will try to rework this.